test/functional/lib/tests.cpp

```diff
@@ -3551,6 +3551,48 @@ GenericXskWait(
     Timer.ExpectElapsed(std::chrono::milliseconds(WaitTimeoutMs));
 }
 
+VOID
+GenericXskWake(
+    _In_ BOOLEAN Rx,
+    _In_ BOOLEAN Tx
+    )
+{
+    auto If = FnMpIf;
+    auto Xsk = SetupSocket(If.GetIfIndex(), If.GetQueueId(), TRUE, TRUE, XDP_GENERIC);
+    auto GenericMp = MpOpenGeneric(If.GetIfIndex());
+    const UINT32 WaitTimeoutMs = 1000;
+
+    XSK_NOTIFY_FLAGS NotifyFlags = XSK_NOTIFY_FLAG_NONE;
+    XSK_NOTIFY_RESULT_FLAGS NotifyResult;
+    if (Rx) {
+        NotifyFlags |= XSK_NOTIFY_FLAG_WAIT_RX;
+    }
+    if (Tx) {
+        NotifyFlags |= XSK_NOTIFY_FLAG_WAIT_TX;
+    }
+
+    auto AsyncThread = std::async(
+        std::launch::async,
+        [&] {
+            //
+            // On another thread, briefly delay execution to give the main test
+            // thread a chance to begin waiting. Then, wake the test thread.
+            //
+            Sleep(100);
+
+            XSK_NOTIFY_RESULT_FLAGS WakeNotifyResult;
+            TEST_HRESULT(XskNotifySocket(Xsk.Handle.get(), XSK_NOTIFY_FLAG_WAKE, 0, &WakeNotifyResult));
+            TEST_EQUAL(WakeNotifyResult, XSK_NOTIFY_RESULT_FLAG_NONE);
+        }
+    );
+
+    //
+    // Verify the IO succeeded (didn't time out).
+    //
+    TEST_HRESULT(XskNotifySocket(Xsk.Handle.get(), NotifyFlags, WaitTimeoutMs, &NotifyResult));
```

Comment:
This still timed out when I used `Sleep(10)` above, so I've updated it to `Sleep(100)` just to make sure, but I think I'm not doing something correct in the product code. @mtfriesen any suggestions?

Reply:
Presumably fixed by 8c6ad8a.